src/graph.cpp

```diff
@@ -165,9 +165,66 @@ bool Graph::path(const std::list<std::string> &source, const std::string &target
     return false;
 }
 
+std::list<Edge *> Graph::bellmanFord(const std::string &source, const std::string &target) {
```

Comment:
Wouldn't a Dijkstra be enough, since we have no negative edge cost and/or capacity? If it works, i guess it doesn't matter much. The complexity difference ain't that much ig

Reply:
It wouldn't work because the for the 4 edges between two vertexes in the residual, the ones that represent removing flow have negative cost (and also create negative cycles)